matchzoo/generators/point_generator.py

```diff
@@ -87,45 +84,17 @@ def _get_batch_of_transformed_samples(
         :param index_array: a list of instance ids.
         :return: A batch of transformed samples.
         """
-        batch_x = {}
-        batch_y = None
-
-        columns = self._left.columns.values.tolist() + \
-            self._right.columns.values.tolist() + ['id_left', 'id_right']
-        for column in columns:
-            batch_x[column] = []
-
-        # Create label field.
-        if self.stage in ['train', 'evaluate']:
-            if isinstance(self._task, tasks.Ranking):
-                self._relation['label'] = self._relation['label'].astype(
-                    self._task.output_dtype)
-                batch_y = self._relation['label'][index_array].values
-            elif isinstance(self._task, tasks.Classification):
-                self._relation['label'] = self._relation['label'].astype(
-                    self._task.output_dtype)
-                batch_y = np.zeros((len(index_array), self._task.num_classes))
-                for idx, label in enumerate(
-                        self._relation['label'][index_array]):
-                    batch_y[idx, label] = 1
-            else:
-                msg = f"{self._task} is not a valid task type."
-                msg += ":class:`Ranking` and :class:`Classification` expected."
-                raise ValueError(msg)
-        # Get batch of X.
-        id_left = self._relation.iloc[index_array, 0]
-        id_right = self._relation.iloc[index_array, 1]
-
-        batch_x['id_left'] = id_left
-        batch_x['id_right'] = id_right
-
-        for column in self._left.columns:
-            batch_x[column] = self._left.loc[id_left, column].tolist()
-        for column in self._right.columns:
-            batch_x[column] = self._right.loc[id_right, column].tolist()
-
-        for key, val in batch_x.items():
-            batch_x[key] = np.array(val)
-        batch_x = utils.dotdict(batch_x)
-
-        return (batch_x, batch_y)
+        sliced_data = self._datapack[index_array]
+        if self._datapack.stage == 'train':
```

Comment:
To address the question raised by @bwanglzu , this line can be if self._datapack.stage in [`train`, `evaluate`]: